main/src/ca/uwaterloo/flix/language/phase/EffectVerifier.scala

```diff
@@ -79,6 +79,11 @@ object EffectVerifier {
       val expected = Type.mkUnion(Type.eraseTopAliases(exp.tpe).arrowEffectType :: exp.eff :: exps.map(_.eff), loc)
       val actual = eff
       expectType(expected, actual, loc)
+    case Expr.ApplyDef(_, exps, _, _, eff, loc) =>
+      exps.foreach(visitExp)
+      val expected = Type.mkUnion(exps.map(_.eff), loc)
```

Comment:
This seems incorrect. Presumably the def itself can have effects?

Reply:
Oh yeah we call pull it off ftpe